sdp/processors/modify_manifest/data_to_data.py

```diff
@@ -14,13 +14,232 @@
 
 import collections
 import re
+import os
 from typing import Dict, List
+import soundfile as sf
 
 from sdp.logging import logger
 from sdp.processors.base_processor import BaseParallelProcessor, DataEntry
+from sdp.utils.common import ffmpeg_convert
 from sdp.utils.edit_spaces import add_start_end_spaces, remove_extra_spaces
 from sdp.utils.get_diff import get_diff_with_subs_grouped
 
+    
+class AudioDuration(BaseParallelProcessor):
+    """
+    Count audio duration using audio file path from input_field
+
+    Args:
+        input_field (str): where to get path to wav file.
+        output_field (str): where to put to audio duration.
+    Returns:
+        All the same fields as in the input manifest plus output_field
+    """
+    def __init__(
+        self,
+        input_field: str,
+        output_field: str,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.input_field = input_field
+        self.output_field = output_field
+    
+    def process_dataset_entry(self, data_entry):
+        audio_filepath = data_entry[self.input_field]
+        try:
+            data, samplerate = sf.read(audio_filepath)
+            data_entry[self.output_field]=data.shape[0]/samplerate
+        except Exception as e:
+            logger.warning(str(e) + " file: " + audio_filepath)
+            data_entry[self.output_field] = -1.0
+        return [DataEntry(data=data_entry)]
+    
+
+class FfmpegConvert(BaseParallelProcessor):
+    """
+    A class for converting video files to audio using FFmpeg and updating the dataset with the path to the resampled audio.
+
+    Args:
+    - resampled_audio_dir (str): The directory to store the resampled audio files.
```

Comment:
Note that none of the new processors will currently show up in the docs. It would be good to add them. Also, I'm not sure that Sphinx will shows these args/methods/notes correctly (need to get rid of the dashes I think)

Reply:
Added, can we check it?

Reply:
fixed it